src/main/java/Erika.java

```diff
@@ -0,0 +1,162 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+
+public class Erika {
+    private static ArrayList<Task> tasks = new ArrayList<>();
+    private static Scanner in = new Scanner(System.in);
+    private static int markIndex;
+    
+    public static void main(String[] args) {
+        printWelcomeMessage();
+        while (true) {
+            String line = collectUserInput();
+            if (line.equals("bye")) {
+                printGoodbyeMessage();
+                break;
+            } else if (line.equals("list")) {
+                printList();
+            } else if (line.contains("mark ")) {
+                handleMark(line);
+            } else if (line.contains("todo ")) {
+                addTodo(line);
+            } else if (line.contains("deadline ")) {
+                addDeadline(line);
+            } else if (line.contains("event ")) {
+                addEvent(line);
+            } else {
+                printUnknownCommand();
+            }
+        }
+    }
+
+    private static void addEvent(String line) {
```

Comment:
The specific implementation of this method (and other methods specific to their task type, e.g. addDeadline(), addTodo()) should be put in the corresponding class to improve the effect of OOP in your code.